src/mdev.rs

```diff
@@ -159,8 +159,14 @@ impl MDev {
                     }
                     // get the key and value from the first (only) map entry
                     if let Some((key, val)) = attrobj.iter().next() {
-                        let valstr = val.as_str().unwrap();
-                        self.attrs.push((key.to_string(), valstr.to_string()));
+                        if let Some(valstr) = val.as_str() {
+                            self.attrs.push((key.to_string(), valstr.to_string()));
+                        } else {
+                            return Err(anyhow!(
+                                "invalid JSON format for attribute {{{:?}, {}}}: value must be of type str",
+                                key, val
+                            ));
+                        }
```

Comment:
This is fine, but I'd personally prefer something like 
```suggestion
                        let valstr = val.as_str().ok_or_else(|| anyhow!(...))?;
                        self.attrs.push((key.to_string(), valstr.to_string()));
```

Reply:
Changed.